storage/backup/backup-cli/src/utils/trace_err.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use diem_logger::error;
+use std::fmt::Display;
+
+pub(crate) trait TraceError<T, E: Display> {
+    fn trace_err(self, notes: &str) -> Result<T, E>;
+}
+
+impl<T, E: Display> TraceError<T, E> for Result<T, E> {
+    fn trace_err(self, notes: &str) -> Result<T, E> {
+        if let Err(e) = &self {
+            error!(error = %e, notes = notes, "Error traced.");
```

Comment:
one word of caution here is that this is the only line number that will show up for these errors (I mean you'll have some form of a stack trace as well)

Reply:
Ugh. I see. Thanks!
The `notes` will definitely help, and hopefully the stack I can relate it to the caller even in an async environment.

Reply:
K, added file!() and line!() as context when necessary.